modules/UI.js

```diff
@@ -1,109 +1,151 @@
+    localStorage.setItem('todos', JSON.stringify(todos))
   }
 
   static clearField = () => {
-    userInput.value = '';
+    userInput.value = ''
   }
 
   static onFocus = (e) => {
-    const ArrayInput = Array.from(document.querySelectorAll('.text'));
+    const ArrayInput = Array.from(document.querySelectorAll('.text'))
     if (e.target.classList.contains('text')) {
-      e.stopPropagation();
+      e.stopPropagation()
       ArrayInput.forEach((item) => {
         if (item.parentElement.parentElement.classList.contains('selected')) {
-          const li = item.parentElement.parentElement;
-          li.classList.remove('selected');
-          li.children[1].textContent = 'more_vertical';
-          li.children[1].classList.remove('delete');
+          const li = item.parentElement.parentElement
+          li.classList.remove('selected')
+          li.children[1].textContent = 'more_vertical'
+          li.children[1].classList.remove('delete')
         }
-      });
+      })
       if (e.target.classList.contains('text')) {
-        e.target.parentElement.parentElement.classList.add('selected');
-        const li = e.target.parentElement.parentElement;
-        li.children[1].textContent = 'delete_forever';
-        li.children[1].classList.add('delete');
+        e.target.parentElement.parentElement.classList.add('selected')
+        const li = e.target.parentElement.parentElement
+        li.children[1].textContent = 'delete_forever'
+        li.children[1].classList.add('delete')
         li.children[1].addEventListener('click', (e) => {
           if (li.children[1].classList.contains('delete')) {
-            UI.deleteTodo(e);
+            UI.deleteTodo(e)
           }
-        });
-        const inputValue = e.target.value;
+        })
+        const inputValue = e.target.value
         e.target.addEventListener('change', () => {
-          const newInput = e.target.value;
-          UI.editText(newInput, inputValue);
-        });
+          const newInput = e.target.value
+          UI.editText(newInput, inputValue)
+        })
       }
     }
   }
+
+  static checkBox = (e) => {
+    console.log(e.target.classList.contains('check'))
+    if (e.target.classList.contains('check')) {
+      let box = e.target.parentElement.children[0]
+      let description = e.target.parentElement.children[1]
+      box.addEventListener('change', () => {
+        if (box.checked) {
+          description.classList.add('strike')
+          Check.check(true, description.value)
+        } else {
+          description.classList.remove('strike')
+          Check.check(false, description.value)
+        }
+      })
+      console.log(e.target.parentElement.children[1].value)
+      console.log(e.target.parentElement.children[0])
+    }
+  }
+
+  static refresh = () =>{
+
+  }
+
+  static clearCompleted = ()=>{
+    let array = getStorage();
+    let filter = array.filter(check=> !check.completed === true)
```

Comment:
you have done a great job implementing all compulsory requirements 

[optional] your completed tasks deletes when the page refreshes, it should present the completed tasks and not delete them upon refresh. the filter function should be called only with a click or mouseDown event listener. Kindly make sure you implement these changes.